### DISCUSSION TITLE
Shader uniforms

### DISCUSSION THREAD

**Discussion by st0rmbtw** [LukasBanana/LLGL#118] [Q&A] [answered]
I have a glsl shader that takes a uniform, for example:
```glsl
uniform mat4 u_view_projection;
```
How can I set that uniform?

With OpenGL I would do something like this:
```cpp
int location = glGetUniformLocation(shader_id, "u_view_projection");
float matrix[] = {  ...  };
glUniformMatrix4fv(location, 1, GL_FALSE, matrix);
```

**Comment by LukasBanana** [comment, marked as answer]
Take a look at the [ResourceBinding](https://github.com/LukasBanana/LLGL/blob/master/examples/Cpp/ResourceBinding/Example.cpp) example and search for "uniform". The GLSL shader use two uniforms, [`instance`](https://github.com/LukasBanana/LLGL/blob/master/examples/Cpp/ResourceBinding/Example.vert#L10) and [`lightVec`](https://github.com/LukasBanana/LLGL/blob/master/examples/Cpp/ResourceBinding/Example.frag#L5). They have to be declared in the `PipelineLayout` and then set via `CommandBuffer::SetUniform` each time the respective PSO is bound.
